### PR TITLE
Add Python Nexmark tests to Jenkins

### PR DESCRIPTION
**Pull request by apilloud**
Run some Batch Nexmark on Python!

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
job_PostCommit_Python_Nexmark_

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
R: @kileys

File: .test-infra/jenkins/job_PostCommit_Python_Nexmark_Direct.groovy [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import CommonTestProperties.SDK
+import CommonTestProperties.TriggeringContext
+import NexmarkBuilder as Nexmark
+import NoPhraseTriggeringPostCommitBuilder
+import PhraseTriggeringPostCommitBuilder
+import InfluxDBCredentialsHelper
+
+import static NexmarkDatabaseProperties.nexmarkBigQueryArgs
+import static NexmarkDatabaseProperties.nexmarkInfluxDBArgs
+
```
Comment (kileys) [inline .test-infra/jenkins/job_PostCommit_Python_Nexmark_Direct.groovy:30]: Unused?

File: .test-infra/jenkins/NexmarkBuilder.groovy [hunk 2]
```diff
@@ -174,10 +181,79 @@ class NexmarkBuilder {
...
+      options.remove('input')
+      options.put('generateEventFilePathPrefix', eventFile)
+
+      gradle {
+        rootBuildScriptDir(commonJobProperties.checkoutDir)
+        tasks(':sdks:java:testing:nexmark:run')
+        commonJobProperties.setGradleSwitches(delegate)
+        switches("-Pnexmark.runner=:runners:direct-java")
+        switches("-Pnexmark.args=\"${parseOptions(options)}\"")
+      }
```
Comment (kileys) [inline .test-infra/jenkins/NexmarkBuilder.groovy:224]: Can we just move this into the run task for python?

File: .test-infra/jenkins/job_PostCommit_Python_Nexmark_Direct.groovy (hunk 1 above)
Comment (apilloud) [reply .test-infra/jenkins/job_PostCommit_Python_Nexmark_Direct.groovy]: Yep, it seems we don't have unused import detection on groovy. Removed from all the nexmark jobs.

File: .test-infra/jenkins/NexmarkBuilder.groovy (hunk 2 above)
Comment (apilloud) [reply .test-infra/jenkins/NexmarkBuilder.groovy:224]: Moving this into the python task is non-trivial, it is running a Java job with different properties than the python job. Now that Java transforms are supported in Python, we would be better off using that to just include the Nexmark Java event generator in the Python job graph. Either option requires significant work, I'm not going to do it here.

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Comment by apilloud** [comment]
Run Seed Job

**Comment by apilloud** [comment]
Run Python Direct Runner Nexmark Tests

**Review by kileys — APPROVED** [review]

**Comment by apilloud** [comment]
Run PythonLint PreCommit

**Comment by apilloud** [comment]
Run RAT PreCommit

**Comment by apilloud** [comment]
Run Python_Integration PreCommit

**Comment by apilloud** [comment]
Run Spotless PreCommit

**Pull request merged**